Mental Medication Additions

BurgerLUA opened the pull request:
# The less controversial tweaks
Tweaked the metabolism rate of mental medication to reflect their intended values. Reduced the dosage threshold to suppress traumas to reflect their intended values. 

Syringe (drugs) now contains truth serum.

Cardox is now slightly poisonous, and can directly remove phoron from blood when consumed. Cardox can now remove phoron in the air when applied to turfs.

A secure box of loyalty implants, hextrasenil pills, and cardox grenades are now located in the vault.

The vault now contains some misc emergency gear that a head of staff can access in case of a dire situation. Current gear are loyalty implants, expensive cardox grenades, and Hextrasenil pills.

## Syndites and Hextrasenil
https://docs.google.com/document/d/1U75T6ytSt2orHpkIKJmY2wPZ61cfpKNY3piJJ8_9-gs/edit?usp=sharing

## WIP
Feedback is need, and balance obviously needs to be done as adding a method to non-surgically remove a loyalty implant is pretty severe. There may need to be proper corporate laws and regulations regarding the appropriate uses of prescribing mind-altering medication to prisoners.

Thread: https://forums.aurorastation.org/viewtopic.php?f=18&t=10968

BurgerLUA commented:
Thread: https://forums.aurorastation.org/viewtopic.php?f=18&t=10968

OneOneThreeEight commented:
No. These are god awful.

Pacmandevil commented:
I like there being a way to disable Loyalty implants - though I'm not sure how much sense this makes in terms of muh lore.

I suspect that anything that can remove an antag status will be metagamed and abused to shit, as is tradition. It might be better to change, or remove that aspect of it.

Pacmandevil commented:
Perhaps have the loyalty implant only disabled while chems above the OD threshold are in the system?

BurgerLUA commented:
@Pacmandevil I was thinking of this, honestly, as well as implementing a stage system of progressing messages and effects for all chems.

Lohikar commented on code/game/objects/items/weapons/grenades/chem_grenade.dm:
```diff
@@ -304,6 +305,29 @@
 
+/obj/item/weapon/grenade/chem_grenade/large/phoroncleaner
+	name = "cardox grenade"
+	desc = "A very expensive large grenade filled with cardox. Removes phoron from the air. WARNING: HAZARDOUS TO VAURCA."
```
Kind of awkward phrasing.

Lohikar commented on code/game/objects/items/weapons/storage/lockbox.dm:
```diff
@@ -126,3 +125,15 @@
+	desc = "A high security lockbox containing special mind-altering medication."
+	req_access = list(access_heads)
+
+	New()
```
Use `fill()`.

Lohikar commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -901,45 +902,91 @@
+		if(istype(H))
+			var/datum/antagonist/antag_data = get_antag_data(H.mind.special_role)
+			if(antag_data && (antag_data.flags & ANTAG_IMPLANT_IMMUNE))
+				H << "You feel the corporate tendrils of [current_map.company_name] try to invade your mind!"
```
`to_chat()`.

Lohikar commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -901,45 +902,91 @@
+				H << "You feel the corporate tendrils of [current_map.company_name] try to invade your mind!"
+			else
+				clear_antag_roles(H.mind, 1)
+				H << "<span class='notice'>You feel a surge of loyalty towards [current_map.company_name]! Any negative intentions towards the company have been long forgotten.</span>"
```
`to_chat()`.

Lohikar commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -901,45 +902,91 @@
+						H.adjustBruteLoss(10)
+						qdel(I)
+						break
+			H << "<span class='notice'><font size =3><B>Your loyalty implant has been deactivated. Freedom, at last!</B></font></span>"
```
`to_chat()`.

Lohikar replied:
Also, how the fuck is this reagent deactivating a loyalty implant?

BurgerLUA replied:
In the loyalty implant description, the way loyalty implants work is that the implant sends nanobots into the bloodstream to modify the implated's brain. It even says that it only works if there are nanobots present in the blood system.

I was thinking of making the traitor medicine to be purchasable only, and contain pills of their own nanobots that interfere with implant.

OneOneThreeEight commented:
I detest the idea of chemicals directly interacting with antagonism. It has far more OOC uses than IC uses.

Pacmandevil commented:
Perhaps instead of disabling the antag status entirely, have it "surpressed"? Meaning you're still technically an antagonist, just that you're incapable of being aggressive towards authority while on the drug, or something upon those lines.

Pacmandevil commented:
and if a free antag status is still a thing, and still an issue. something along the lines of tg's "sintouched" antag where it's just minor shit you can do would be better, instead of giving them a free pass for whatever.

BurgerLUA commented:
Added a new reagent that purges chems from the system, should be useful for removing syndites and Hextrasenil, among other things.

Overdose time for conversion is now boosted from 3 minutes to 5 minutes.

Legal antipsychotics now show up on medical scanners.

Added panocelium, a mushroom mutated from fly amanita. which contains panotoxin. Panotoxin is a poison that causes intense pain to whoever consumes it.

I never tested any of this yet btw.

BurgerLUA commented:
Huge Update, yet to be tested.

- Added Inhalers.
- Added the ability for mental medications to add traumas permanently while the medication is active (for desired traumas.)
- Added the breath analyzer, monitors oxyloss, lung damage, intoxication, and medical reagents in the lungs.
- Replaced oxy first aid kit contents with inhalers containing dexalin and breath analyzers.
- Replaced the combat medkit spaceacillin pills with an inhaler of hyperzine
- Added two chloral hydrate inhalers to surgery kits
- Added space drugs inhaler to random contraband loot
- Added hyperzine inhaler to random maint loot (rare)
- Added breath analyzers to the warden's locker, the head of security's locker,
- Added random inhalers to cargo loot, along with breath analyzers
- Added strength multiplier effects for reagents based on method of consumption
- Ammonia is twice as poisonous when inhaled.
- Breathing in acid now deals organ damage to the lungs.
- Breathing in bicardine slowly heals the lungs, if the lungs are not completely damaged.
- Dexalin and dexalin plus are twice as powerful when inhaled.
- Hyperzine lasts twice as long when inhaled.
- Added feartoxin, a toxin that grants a temporary phobia and hallucinations when injected.
- Liquid phoron is twice as potent when inhaled.
- Sophoric is twice as potent when inhaled.
- Chloral Hydrate is twice as potent when inhaled.
- Space drugs are twice as potent when inhaled.

BurgerLUA commented:
I went insane again
- Reworked Intoxication because it was easier than coming up with a custom complex formula.
- Slurring is now linear.

LordFowl commented:
This PR is quickly become the very opposite of atomic. If I were you, I would consider moving some of these additions to separate PRs, and stop piling on to this one.

BurgerLUA commented:
There are some things that rely entirely on others, such as the breath analyzer for inhalers relying on intoxication, and the intoxication relying on mental medication.

LordFowl commented:
In that case you should, particularly in future PRs, consider patience. A PR for one thing should stop at that one thing, and not try to add in a hundred other things only loosely correlated.

BurgerLUA commented:
I wanted to add inhalers drugs. By adding inhaler drugs, I needed to add breath analyzers. I wanted to make breath analyzers that measure alcohol content, but the current system was horseshit so I redid that.

OneOneThreeEight commented:
It wouldn't hurt to rehash the title of this PR, its contents and then make a new thread with all of the revisions. As it stands, all previous discussion is old news at this point.

BurgerLUA commented:
Todo:
- Finally Tweak Intoxication to a working level
- Check if you can grow/mutate/harvest panocelium
- Check if hex and syndites actually fucking work
- Test Calomel 
- Test cardox

Arrow768 commented:
Indeed, Shev is correct.
I would even suggest to close this PR and open a new one with a correct title

BurgerLUA commented:
I strongly disagree. Intoxication is mostly the same, just codewise it's improved.

BurgerLUA commented:
I'll just make a new feedback thread, the other was was shat up anyways. I just pray to god this is actually merged because the poll seems good.

BurgerLUA commented:
https://forums.aurorastation.org/viewtopic.php?f=18&t=11074&p=98506#p98506

OneOneThreeEight commented:
I'm rescinding whatever opposition I held previously. Not unlike the man who is quoted to have 'gotten better' after being turned into a newt by a witch, this PR also 'got better.'

It's less concerning that hextrasenil exists *though I really think it's slated for a better name than currently since it's too tongue-in-cheek* given the consideration that the aforementioned chem is considered under Directive 4, and an experimental chem at that.

I don't necessarily mind the presence of syndites (also due for a better name) at this point, but the mechanics in regards to how it influences people is also not very tongue-in-cheek, given that it lampshades the same way that mind-slave implants from /tg/ do.

BurgerLUA commented:
its finally fucking done

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,96 @@ REAGENT SCANNER
+		user.show_message("<span class='danger'>Alert: No breathing detected.</span>")
+		return
+
+	if (!usr.IsAdvancedToolUser())
```
`usr` in a non-public interface. No. Replace with `user`.

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,96 @@ REAGENT SCANNER
+		return
+
+	if (!usr.IsAdvancedToolUser())
+		usr << "<span class='warning'>You don't have the dexterity to do this!</span>"
```
`usr` => `user`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,96 @@ REAGENT SCANNER
+		usr << "<span class='warning'>You don't have the dexterity to do this!</span>"
+		return
+
+	if(user == H)
```
Pointless pyramid.
`if (user == H && !H.can_eat(src))`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,96 @@ REAGENT SCANNER
+	if(user == H)
+		if(!H.can_eat(src))
+			return
+	else
```
`else if (!H.can_force_feed(user, src))`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,96 @@ REAGENT SCANNER
+
+	user.show_message("<b>Breath Sample Results:</b>")
+
+	if(H.stat == 2 || H.losebreath || !H.breathing)
```
Use a define for that stat variable. I assume you mean `H.stat == DEAD`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,203 @@
+				update_icon()
+		return
+
+	if (!usr.IsAdvancedToolUser())
```
`usr` => `user`

BurgerLUA commented:
ITS GUCCI

skull132 commented on code/modules/mob/living/carbon/carbon_defines.dm:
```diff
@@ -17,6 +17,7 @@
 	var/pose = null
 	var/list/chem_effects = list()
 	var/intoxication = 0//Units of alcohol in their system
+	var/bac = 0 //Blood alcohol level, as calculated inside intoxication.dm
```
Oh nonono. No. Delete this and replace with a proc.

Retaining this state will lead to horrible horrible things. A stateless calculation each time it's needed is better. Also, the name is absoloutely horrible.

Also consider making the proc a `human` proc.

skull132 commented on code/modules/mob/mob_helpers.dm:
```diff
@@ -288,26 +288,27 @@ var/list/global/organ_rel_size = list(
+				if(1,3,5,8)	newletter="[lowertext(newletter)]"
+				if(2,4,6,15)	newletter="[uppertext(newletter)]"
+				if(7)	newletter+="'"
+				//if(9,10)	newletter="<b>[newletter]</b>"
```
Delete this and the two lines below.

skull132 commented on code/modules/reagents/reagent_containers/inhaler.dm:
```diff
@@ -0,0 +1,189 @@
+			user.show_message("<span class='notice'>[trans] units injected. [reagents.total_volume] units remaining in \the [src].</span>")
+		return
+
+	if (!usr.IsAdvancedToolUser())
```
`usr` => `user`

skull132 commented on code/modules/reagents/reagent_containers/inhaler.dm:
```diff
@@ -0,0 +1,189 @@
+		return
+
+	if (!usr.IsAdvancedToolUser())
+		usr << "<span class='warning'>You don't have the dexterity to do this!</span>"
```
`usr` => `user`

skull132 commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -524,92 +540,101 @@
 /datum/reagent/mental/affect_blood(var/mob/living/carbon/human/H, var/alien, var/removed)
 
-	if(!istype(H) || world.time < data)
+	if(!istype(H) || max_dose < min_dose || (world.time < data && volume > removed) || messagedelay == -1)
```
`max_dose < min_dose` What is this? It seems like this is trying to enforce compile time mistakes at run time?  These are variables which shouldn't be changed at run time, so checking them isn't necessary. Remove.

Same here. What is this? `volume > removed`

BurgerLUA replied:
max_dose < min_dose and volume > removed prevents message startup/startdown spam if you're using something like a cigarette or similiar device that transfers a little bit of reagents at a time.
f in the case that a cigarette has a lower transfer rate than the metabolism rate, then message spam will occur since it's starting and stopping constantly.
This also prevents the whole code from working if the dosage is very small.

Lohikar commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+
+/obj/item/device/breath_analyzer
+	name = "breath analyzer"
+	desc = "A hand-held breath analyzer that provides a robust amount of information about the subject's respitory system."
```
`respitory`

Lohikar commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+
+	if(user == H && !H.can_eat(src))
+		return
+	else if(!H.can_force_feed(user, src))
```
No messages for these?

BurgerLUA replied:
Messages are handled inside the procs can_eat and can_force_feed.

Lohikar commented on code/modules/mob/living/carbon/human/intoxication.dm:
```diff
@@ -37,58 +21,72 @@ var/mob/living/carbon/human/alcohol_clumsy = 0
+			if(bac > INTOX_CONSCIOUS*SR)
+				slurring = max(slurring, 90)
+				src.visible_message("<span class='danger'>[src] loses consciousness!</span>","<span class='danger'>You lose consciousness!</span>")
+				paralysis = max(paralysis, 60 SECONDS)
```
Isn't there a proc for setting this?

Alberyk replied:
Paralyse().

skull132 commented on code/modules/mob/living/carbon/human/intoxication.dm:
```diff
@@ -37,58 +21,72 @@ var/mob/living/carbon/human/alcohol_clumsy = 0
+	src.bac = round(intoxication/max(vessel.get_reagent_amount("blood"),1),0.01)
+
+	if(bac > INTOX_BUZZED*SR && bac < INTOX_MUSCLEIMP*SR && !(locate(/datum/modifier/buzzed) in modifiers))
+		src.show_message("<span class='notice'>You feel buzzed.</span>")
```
You mean to use `to_chat(src, "message")`. `show_message` has unintended side-effects. Specifically, when unconscious, you get this message: `"<I>... You can almost hear someone talking ...</I>"`. Which is not befitting a status message.

skull132 commented on code/modules/mob/living/carbon/human/intoxication.dm:
```diff
@@ -37,58 +21,72 @@ var/mob/living/carbon/human/alcohol_clumsy = 0
-			src << "<span class='notice'>The room starts spinning!</span>"
-		var/target_dizziness = min(1000,(BASE_DIZZY + ((intoxication - AE_DIZZY*SR)*10)/SR))
-		make_dizzy(target_dizziness - dizziness) // We will repeatedly set our target dizziness to a desired value based on intoxication level
+			src.show_message("<span class='notice'>You feel a little tipsy.</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/mob/living/carbon/human/intoxication.dm:
```diff
@@ -37,58 +21,72 @@ var/mob/living/carbon/human/alcohol_clumsy = 0
+	if(bac > INTOX_MUSCLEIMP*SR)
+		slurring = max(slurring, 25)
+		if (!(locate(/datum/modifier/drunk) in modifiers))
+			src.show_message("<span class='notice'>You feel drunk!</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/mob/living/carbon/human/intoxication.dm:
```diff
@@ -37,58 +21,72 @@ var/mob/living/carbon/human/alcohol_clumsy = 0
-	//Make the drinker temporarily clumsy if intoxication is high enough
-	//We use a var to track if alcohol caused it, we won't add nor remove it if the drinker was already clumsy from some other source
-	if(intoxication > AE_CLUMSY*SR)
+			src.show_message("<span class='warning'>You feel uncoordinated and unsteady on your feet!</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/mob/living/carbon/human/intoxication.dm:
```diff
@@ -37,58 +21,72 @@ var/mob/living/carbon/human/alcohol_clumsy = 0
-		if (life_tick % 4 == 1)//Only process vomit chance periodically
-			var/chance = BASE_VOMIT_CHANCE + ((intoxication - AE_VOMIT)*VOMIT_CHANCE_SCALE)
+	else if (alcohol_clumsy)
+		src.show_message("<span class='notice'>You feel more sober and steady.</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/mob/living/carbon/human/intoxication.dm:
```diff
@@ -37,58 +21,72 @@ var/mob/living/carbon/human/alcohol_clumsy = 0
+				adjustBrainLoss(3,10)
+			else if(prob(10))
+				slurring = max(slurring, 70)
+				src.show_message("<span class='warning'>You decide that you like the ground and spend a few seconds to rest.</span>")
```
`show_message` => `to_chat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+	var/mob/living/carbon/human/H = M
+
+	if (!istype(H))
+		user.show_message("<span class='warning'>You can't find a way to use the [src] on [M].</span>")
```
`show_message` => `to_chat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+		return
+
+	if ( ((CLUMSY in user.mutations) || (DUMB in user.mutations)) && prob(20))
+		user.show_message("<span class='danger'>Your hand slips from clumsiness!</span>")
```
`show_message` => `to_chat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+	if ( ((CLUMSY in user.mutations) || (DUMB in user.mutations)) && prob(20))
+		user.show_message("<span class='danger'>Your hand slips from clumsiness!</span>")
+		eyestab(M,user)
+		user.show_message("<span class='danger'>Alert: No breathing detected.</span>")
```
`show_message` => `to_chat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+		return
+
+	if (!user.IsAdvancedToolUser())
+		user.show_message("<span class='warning'>You don't have the dexterity to do this!</span>")
```
`show_message` => `to_chat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+		src.other_DNA_type = "saliva"
+
+	if (!do_after(user, 2 SECONDS, act_target = M))
+		user.show_message("<span class='notice'>You and the target need to be standing still in order to take a breath sample.</span>")
```
`show_message` => `to_chat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+	user.show_message("<b>Breath Sample Results:</b>")
+
+	if(H.stat == DEAD || H.losebreath || !H.breathing)
+		user.show_message("<span class='danger'>Alert: No breathing detected.</span>")
```
`show_message` => `to_chat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+
+	user.visible_message("<span class='notice'>[user] takes a breath sample from [M].</span>","<span class='notice'>The [src] clicks as it finishes reading [M]'s breath sample.</span>")
+
+	user.show_message("<b>Breath Sample Results:</b>")
```
`show_message` => `to_chat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+		user.show_message("<span class='danger'>Alert: No breathing detected.</span>")
+		return
+
+	user.show_message(H.getOxyLoss() > 50 ? "<font color='blue'><b>Severe oxygen deprivation detected.</b></font>" : "Subject oxygen levels normal.")
```
`show_message` => `to_chat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+	user.show_message(H.getOxyLoss() > 50 ? "<font color='blue'><b>Severe oxygen deprivation detected.</b></font>" : "Subject oxygen levels normal.")
+	var/obj/item/organ/L = H.internal_organs_by_name["lungs"]
+	if(istype(L))
+		user.show_message(L.is_bruised() ? "<font color='red'><b>Ruptured lung detected.</b></font>" : "Subject lung health normal.")
```
`show_message` => `to_chat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+	if(istype(L))
+		user.show_message(L.is_bruised() ? "<font color='red'><b>Ruptured lung detected.</b></font>" : "Subject lung health normal.")
+	else
+		user.show_message("<span class='warning'>Subject lung health unknown.</span>")
```
`show_message` => `to_chat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+			additional_string = "<font color='red'>\[ALCOHOL POISONING LIKELY\]</font>"
+		if(INTOX_DEATH to INFINITY)
+			additional_string = "<font color='red'>\[DEATH IMMINENT\]</font>"
+	user.show_message("<span class='normal'>Blood Alcohol Content: [round(H.bac,0.01)] <b>[additional_string]</b></span>")
```
`show_message` => `to_chat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+		var/unknown = 0
+		for(var/datum/reagent/R in H.breathing.reagent_list)
+			if(R.scannable)
+				user.show_message("<span class='notice'>[R.name] found in subject's respitory system.</span>")
```
`show_message` => `to_chat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,94 @@ REAGENT SCANNER
+			else
+				++unknown
+		if(unknown)
+			user.show_message("<span class='warning'>Non-medical reagent[(unknown > 1)?"s":""] found in subject's respitory system.</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler.dm:
```diff
@@ -0,0 +1,189 @@
+/obj/item/weapon/reagent_containers/inhaler/attack(mob/living/M as mob, mob/user as mob)
+
+	if(!reagents.total_volume)
+		user.show_message("<span class='warning'>\The [src] is empty.</span>")
```
aaaaaaaaaa `show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler.dm:
```diff
@@ -0,0 +1,189 @@
+		return
+
+	if ( ((CLUMSY in user.mutations) || (DUMB in user.mutations)) && prob(10))
+		user.show_message("<span class='danger'>Your hand slips from clumsiness!</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler.dm:
```diff
@@ -0,0 +1,189 @@
+			admin_inject_log(user, M, src, contained, trans)
+			playsound(src.loc, 'sound/items/stimpack.ogg', 50, 1)
+			user.visible_message("<span class='notice'>[user] accidentally sticks the [src] in [M]'s eye and presses the injection button!</span>","<span class='notice'>You accidentally stick the [src] in [M]'s eye and press the injection button!</span>")
+			user.show_message("<span class='notice'>[trans] units injected. [reagents.total_volume] units remaining in \the [src].</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler.dm:
```diff
@@ -0,0 +1,189 @@
+	else
+		user.visible_message("<span class='warning'>[user] attempts to administer \the [src] to [M]...</span>","<span class='notice'>You attempt to administer \the [src] to [M]...</span>")
+		if (!do_after(user, 1 SECONDS, act_target = M))
+			user.show_message("<span class='notice'>You and the target need to be standing still in order to inject \the [src].</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler.dm:
```diff
@@ -0,0 +1,189 @@
+		var/trans = reagents.trans_to_mob(M, amount_per_transfer_from_this, CHEM_BREATHE)
+		admin_inject_log(user, M, src, contained, trans)
+		playsound(src.loc, 'sound/items/stimpack.ogg', 50, 1)
+		user.show_message("<span class='notice'>[trans] units injected. [reagents.total_volume] units remaining in \the [src].</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+
+	if(is_open_container())
+		if(reagents && reagents.reagent_list.len)
+			user.show_message("<span class='notice'>It contains [round(reagents.total_volume, accuracy)] units of non-aerosol mix.</span>")
```
**AAAAAAAAAAAAAAAAAAAA** `show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+		if(reagents && reagents.reagent_list.len)
+			user.show_message("<span class='notice'>It contains [round(reagents.total_volume, accuracy)] units of non-aerosol mix.</span>")
+		else
+			user.show_message("<span class='notice'>It is empty.</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+		else
+			user.show_message("<span class='notice'>It is empty.</span>")
+	else
+		user.show_message("<span class='notice'>The reagents are secured in the aerosol mix.</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+
+/obj/item/weapon/reagent_containers/personal_inhaler_cartridge/attack_self(mob/user as mob)
+	if(is_open_container())
+		user.show_message("<span class='notice'>With a quick twist of the lid, you secure the reagents inside \the [src].</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+		user.show_message("<span class='notice'>With a quick twist of the lid, you secure the reagents inside \the [src].</span>")
+		flags ^= OPENCONTAINER
+	else
+		user.show_message("<span class='notice'>\The reagents inside [src] are already secured.</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+	if(!..(user, 2))
+		return
+	if(stored_cartridge)
+		user.show_message("<span class='notice'>\The [stored_cartridge] is attached to \the [src].</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+/obj/item/weapon/personal_inhaler/attack_self(mob/user as mob)
+	if(stored_cartridge)
+		user.put_in_hands(stored_cartridge)
+		user.show_message("<span class='warning'>You remove \the [stored_cartridge] from \the [src].</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+		return
+
+	if(!stored_cartridge)
+		user.show_message("<span class='warning'>\The [src] has no cartridge installed!</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+		return
+
+	if(!stored_cartridge.reagents || !stored_cartridge.reagents.total_volume)
+		user.show_message("<span class='warning'>\The [src]'s cartridge is empty!</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+		return
+
+	if ( ((CLUMSY in user.mutations) || (DUMB in user.mutations)) && prob(10))
+		user.show_message("<span class='danger'>Your hand slips from clumsiness!</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+			var/trans = stored_cartridge.reagents.trans_to_mob(M, transfer_amount, CHEM_TOUCH)
+			admin_inject_log(user, M, src, contained, trans)
+			user.visible_message("<span class='notice'>[user] accidentally sticks \the [src] in [M]'s eye and presses the injection button!</span>","<span class='notice'>You accidentally stick the [src] in [M]'s eye and press the injection button!</span>")
+			user.show_message("<span class='notice'>[trans] units injected. [reagents.total_volume] units remaining in \the [src].</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+			user.show_message("<span class='notice'>[trans] units injected. [reagents.total_volume] units remaining in \the [src].</span>")
+			playsound(src.loc, 'sound/items/stimpack.ogg', 50, 1)
+			if(eject_when_empty)
+				user.show_message("<span class='notice'>\The [stored_cartridge] automatically ejects from \the [src].</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+		return
+
+	if (!user.IsAdvancedToolUser())
+		user.show_message("<span class='warning'>You don't have the dexterity to do this!</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+	else
+		user.visible_message("<span class='warning'>[user] attempts to administer \the [src] to [M]...</span>","<span class='notice'>You attempt to administer \the [src] to [M]...</span>")
+		if (!do_after(user, 1 SECONDS, act_target = M))
+			user.show_message("<span class='notice'>You and \the [M] need to be standing still in order to inject \the [src].</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+/obj/item/weapon/personal_inhaler/attackby(var/obj/item/weapon/reagent_containers/personal_inhaler_cartridge/cartridge as obj, var/mob/user as mob)
+	if(istype(cartridge))
+		if(src.stored_cartridge)
+			user.show_message("<span class='notice'>\The [src] already has a cartridge.</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+			user.show_message("<span class='notice'>\The [src] already has a cartridge.</span>")
+			return
+		if(cartridge.is_open_container())
+			user.show_message("<span class='notice'>\The [cartridge] needs to be secured first.</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+		user.show_message("<span class='notice'>[trans] units injected. [reagents.total_volume] units remaining in \the [src].</span>")
+		playsound(src.loc, 'sound/items/stimpack.ogg', 50, 1)
+		if(eject_when_empty)
+			user.show_message("<span class='notice'>\The [stored_cartridge] automatically ejects from \the [src].</span>")
```
`show_message` => `to_chat`

skull132 commented on code/modules/reagents/reagent_containers/inhaler_advanced.dm:
```diff
@@ -0,0 +1,201 @@
+		var/contained = stored_cartridge.reagentlist()
+		var/trans = stored_cartridge.reagents.trans_to_mob(M, transfer_amount, CHEM_BREATHE)
+		admin_inject_log(user, M, src, contained, trans)
+		user.show_message("<span class='notice'>[trans] units injected. [reagents.total_volume] units remaining in \the [src].</span>")
```
`show_message` => `to_chat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,92 @@ REAGENT SCANNER
+	to_chat(user,"<b>Breath Sample Results:</b>")
+
+	if(H.stat == DEAD || H.losebreath || !H.breathing)
+		to_chat("<span class='danger'>Alert: No breathing detected.</span>")
```
Missing first argument.

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,92 @@ REAGENT SCANNER
+	if ( ((CLUMSY in user.mutations) || (DUMB in user.mutations)) && prob(20))
+		to_chat(user,"<span class='danger'>Your hand slips from clumsiness!</span>")
+		eyestab(H,user)
+		user.show_message("<span class='danger'>Alert: No breathing detected.</span>")
```
`toc_hat`

skull132 commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -402,4 +393,92 @@ REAGENT SCANNER
+		to_chat("<span class='danger'>Alert: No breathing detected.</span>")
+		return
+
+	user.show_message(H.getOxyLoss() > 50 ? "<font color='blue'><b>Severe oxygen deprivation detected.</b></font>" : "Subject oxygen levels normal.")
```
to_chat

skull132 commented:
I will absoloutely regret this and I will make sure that this doesn't happen again.

But I've reviewed this like, 4 times now and it seems sane. So _fine_.
